e2e/setup/helpers.ts

```diff
@@ -268,16 +268,25 @@ export const waitUntil = async (
   let tries = 0;
   let isConditionMet = false;
 
-  do {
-    isConditionMet = await condition();
-    tries++;
-
-    if (tries > maxTries) {
-      throw new Error("Condition was not met within the time limit");
+  try {
+    do {
+        isConditionMet = await condition();
+        tries++;
+
+        if (tries > maxTries) {
+          throw new Error("Condition was not met within the time limit");
+        }
+
+        await new Promise((resolve) => setTimeout(resolve, timeoutMs));
+      } while (!isConditionMet);
+  } catch (error) {
+    // We know that sometimes network requests can fail, so we retry once more
+    if (isConditionMet === false) {
+      return await waitUntil(condition, maxTries, timeoutMs);
     }
 
-    await new Promise((resolve) => setTimeout(resolve, timeoutMs));
-  } while (!isConditionMet);
+    throw error;
+  }
```

Comment:
I think you said you wanted this to be located in the `condition` function we pass, no?